packages/rs-sdk/src/platform/document_query.rs

```diff
@@ -0,0 +1,300 @@
+//! Method to query documents from the Drive.
+
+use std::sync::Arc;
+
+use crate::{error::Error, sdk::Sdk};
+use ciborium::Value as CborValue;
+use dapi_grpc::platform::v0::{
+    self as platform_proto, get_documents_request::Start, GetDocumentsRequest,
+};
+use dpp::{
+    data_contract::{
+        accessors::v0::DataContractV0Getters, document_type::accessors::DocumentTypeV0Getters,
+    },
+    document::Document,
+    platform_value::{platform_value, Value},
+    prelude::{DataContract, Identifier},
+};
+use drive::query::{DriveQuery, InternalClauses, OrderClause, WhereClause, WhereOperator};
+use drive_proof_verifier::FromProof;
+use rs_dapi_client::transport::{
+    AppliedRequestSettings, BoxFuture, TransportClient, TransportRequest,
+};
+
+use super::fetch::Fetch;
+
+/// Request that is used to query documents from the Dash Platform.
+///
+/// This is an abstraction layer built on top of [GetDocumentsRequest] to address issues with missing details
+/// required to correctly verify proofs returned by the Dash Platform.
+///
+/// Conversions are implemented between this type, [GetDocumentsRequest] and [DriveQuery] using [TryFrom] trait.
+#[derive(Debug, Clone, serde::Serialize)]
+pub struct DocumentQuery {
+    /// Data contract ID
+    pub data_contract: Arc<DataContract>,
```

Comment:
Don't use Arc, we're on the client, so cloning is fine. However I would do this fix in another PR and not this one. As the other PR we could get a good feel about what is being changed.

Reply:
Let's leave ARC for now, we can think on how to better do this later.